tests/mock_vllm_model.py

```diff
@@ -286,24 +272,24 @@ def decode_forward(
         # logits: [batch, seqlen, vocab_size]
         logits = torch.randn((batch, seqlen, 128256))
         # send a token every period loops
-        EOT_ID = 128009
-        # EOS_ID = 128001
-        send_index = 200
-        send_token = EOT_ID
-        if start_pos is not None:
-            if isinstance(start_pos, int):
-                # if start pos is same across batch, ie. now in prefill
-                cache_idxs = torch.tensor(
-                    [start_pos for _ in range(batch)], dtype=torch.int64
-                )
-            else:  # if start_pos is a tensor ie. is different across batch, now in decode mode
-                # if start position is greater than index to send EOT
-                cache_idxs = start_pos.to(dtype=torch.int64)
-                send_token_mask = cache_idxs > send_index
-                # find positions where start pos passes send_index (ie. done decoding) + make 1D
-                batch_indices = torch.nonzero(send_token_mask).squeeze()
-                # assign a high logit at at the send _token index so model will select it and generate the EOT so that generation stops
-                logits[batch_indices, 0, send_token] = 100.0
+        # EOT_ID = 128009
+        # # EOS_ID = 128001
+        # send_index = 200
```

Comment:
nice catch! but, can we have a variable fixed EOR send index based on the max number of tokens allowed when the mock model get initialized? Or does it not matter that there is an EOT at the end of the output sequence for the mock model?

Reply:
It doesn't matter for the mock model, vLLM will cut off the generation as expected. I'm leaving the code block in case we need it to test a specific case where a token gets sent at specific time/output length.

Reply:
That makes sense. Thanks for clarifying!